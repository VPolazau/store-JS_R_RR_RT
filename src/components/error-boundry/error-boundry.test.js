import React from 'react'
import renderer from 'react-test-renderer'

import ErrorBoundary from './error-boundry'

test('test ErrorBoundary', () => {
  const Child = () => {
    throw new Error()
  }

  const component = renderer.create(
    <ErrorBoundary>
      <Child />
    </ErrorBoundary>
  )

  let componentTree = component.toJSON()
  expect(componentTree).toMatchSnapshot()
})


// import { render } from '@testing-library/react'

// const renderProviders = (ui: React.ReactElement) => render(ui, {})

// const Child = () => {
//   throw new Error()
// }

// describe('Error Boundary', () => {
//   it(`should render error boundary component when there is an error`, () => {
//     const { getByText } = renderProviders(
//       <ErrorBoundary>
//         <Child />
//       </ErrorBoundary>
//     )

//     const errorMessage = getByText('something went wrong')
//     expect(errorMessage).toBeDefined()
//   })
// })