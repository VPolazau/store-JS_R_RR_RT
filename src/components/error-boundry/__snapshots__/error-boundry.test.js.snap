// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorBoundary 1`] = `
<div
  className="ErrorIndicator__div"
>
  <span
    className="span-error"
  >
    Error!
  </span>
  <h1
    className="description-error"
  >
    Something went wrong.
  </h1>
</div>
`;
